File: app/src/main/java/com/ccamacho/udemycoursejetpack/views/TodoView.java
package com.ccamacho.udemycoursejetpack.views;

import android.content.Context;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.constraintlayout.widget.ConstraintLayout;

import com.ccamacho.udemycoursejetpack.R;
import com.ccamacho.udemycoursejetpack.models.Todo;

public class TodoView extends ConstraintLayout {

    private CheckBox completeCheckbox;
    private TextView descriptionView;

    public TodoView(Context context) {
        super(context);
    }

    public TodoView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public TodoView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void initView(Todo todo) {
        completeCheckbox = findViewById(R.id.complete_checkBox);
        descriptionView = findViewById(R.id.description_view);

        descriptionView.setText(todo.getDescription());
        completeCheckbox.setChecked(todo.isComplete());

        if (todo.isComplete()) {
            createStrikeThrough();
        }

        setUpCheckStateListener();
    }

    public void setUpCheckStateListener() {
        completeCheckbox.setOnCheckedChangeListener((button, isChecked) -> {
            removeStrikeThrough();
            if (isChecked) {
                createStrikeThrough();
            }
        });
    }

    private void createStrikeThrough() {
        descriptionView.setPaintFlags(descriptionView.getPaintFlags() | Paint.STRIKE_THRU_TEXT_FLAG);
    }

    private void removeStrikeThrough() {
        descriptionView.setPaintFlags(descriptionView.getPaintFlags() & ~(Paint.STRIKE_THRU_TEXT_FLAG));
    }
}

File: app/src/main/java/com/ccamacho/udemycoursejetpack/views/TodoView.kt
package com.ccamacho.udemycoursejetpack.views

import android.content.Context
import android.graphics.Paint
import android.util.AttributeSet
import androidx.constraintlayout.widget.ConstraintLayout
import com.ccamacho.udemycoursejetpack.models.Todo
import kotlinx.android.synthetic.main.view_todo.view.*

class TodoView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 1
) : ConstraintLayout(context, attrs, defStyleAttr) {

    fun initView(todo: Todo, callback: ((Boolean) -> Unit)? = null) {
        description_view.text = todo.description
        complete_checkBox.isChecked = todo.isComplete
        if (todo.isComplete) {
            createStrikeThrough()
        }
        setUpCheckStateListener(todo, callback)
    }

    fun setUpCheckStateListener(todo: Todo, callback: ((Boolean) -> Unit)? = null) {
        complete_checkBox.setOnCheckedChangeListener { _, isChecked: Boolean ->
            todo.isComplete = isChecked
            callback?.invoke(isChecked)
            removeStrikeThrough()
            if (isChecked) {
                createStrikeThrough()
            }
        }
    }

    private fun createStrikeThrough() {
        description_view.apply {
            paintFlags = paintFlags or Paint.STRIKE_THRU_TEXT_FLAG
        }
    }

    private fun removeStrikeThrough() {
        description_view.apply {
            paintFlags = paintFlags and Paint.STRIKE_THRU_TEXT_FLAG.inv()
        }
    }
}